src/providers/tpm/utils.rs

```diff
@@ -105,19 +105,23 @@ pub fn parsec_to_tpm_params(attributes: Attributes) -> Result<KeyParams> {
                 x @ 1024 | x @ 2048 | x @ 3072 | x @ 4096 => x.try_into().unwrap(), // will not fail on the matched values
                 _ => return Err(ResponseStatus::PsaErrorInvalidArgument),
             };
-            if attributes.is_encrypt_permitted() || attributes.is_decrypt_permitted() {
-                Ok(KeyParams::RsaEncrypt {
+            match attributes.policy.permitted_algorithms {
+                Algorithm::AsymmetricSignature(_) => Ok(KeyParams::RsaSign {
```

Comment:
I'm not sure what would happen if this was `Algorithm::AsymmetricSignature(AsymmetricSignature::Ecdsa { ... }`. You'll definitely get an error somewhere, but I'm not sure it'll be `PsaErrorInvalidParameter`. Do we have those sort of tests? Maybe we should 😬

Reply:
What about:
```rust
Algorithm::AsymmetricSignature(alg) if alg.is_rsa() => ...
```
it's already in [`psa-crypto`](https://docs.rs/psa-crypto/0.8.0/psa_crypto/types/algorithm/enum.AsymmetricSignature.html#method.is_rsa_alg) 🤯 

We have [one test](https://github.com/parallaxsecond/parsec/blob/b6b73160498e52cbfb3527f5120036c729f91920/e2e_tests/tests/per_provider/normal_tests/key_attributes.rs#L16) which is a bit like that but is currently ignored 🛩️ Should maybe add more at some point.

Reply:
Yeah, that's pretty neat!!